sdk/servicebus/service-bus/src/models.ts

```diff
@@ -57,11 +73,22 @@ export interface MessageHandlers {
  */
 export interface InternalMessageHandlers extends MessageHandlers {
   /**
-   * Called when the connection is initialized but before we subscribe to messages or add credits.
-   *
+   * Called when the connection is initialized but before we've added credits.
    * NOTE: This handler is completely internal and only used for tests.
    */
-  processInitialize?: () => Promise<void>;
+  postInitialize?: () => Promise<void>;
+
+  /**
+   * Called before we actually initialize the link itself.
+   * NOTE: This handler is completely internal and only used for tests.
+   */
+  preInitialize?: () => Promise<void>;
```

Comment:
Nit: To add more differentiation from the `processInitialize` method, can we use the term "link" here?
`postLinkInitialize` and `preLinkInitialize`? This way `processInitialize` can be continued to be thought of as to be in the context of a partition getting ready to be processed.